test/unit/modules/blocks/chain.js

```diff
@@ -15,313 +15,596 @@
-					it('should call callback with error');
+							it('should call a callback with no error', done => {
+								blocksChainModule.saveGenesisBlock(cb => {
+									expect(cb).to.be.null;
+									done();
+								});
+							});
+						});
+					});
 				});
-			});
 
-			it('should call callback with error = undefined');
+				describe('if returns row', () => {
+					beforeEach(() => {
+						return library.db.blocks.getGenesisBlockId.resolves([{ id: 1 }]);
+					});
 
-			it('should call callback with result = undefined');
+					it('should call a callback with no error', done => {
+						blocksChainModule.saveGenesisBlock(err => {
+							expect(err).to.be.undefined;
+							done();
+						});
+					});
+				});
+			});
 		});
 	});
 
 	describe('saveBlock', () => {
-		it('should call library.db.transaction');
-
-		it('should call library.logic.block.dbSave');
-
-		it('should call library.logic.block.dbSave with block');
-
-		describe('when library.db.transaction callback throws', () => {
-			it('should call logger.error');
+		let afterSaveTemp;
 
-			it('should call logger.error with err.stack');
-
-			it('should call callback with "Blocks#saveBlock error"');
+		beforeEach(done => {
+			afterSaveTemp = __private.afterSave;
+			done();
 		});
 
-		describe('when library.db.transaction callback does not throw', () => {
-			it('should call __private.afterSave');
-
-			it('should call __private.afterSave with block');
-
-			it('should call __private.afterSave with callback');
+		afterEach(done => {
+			__private.afterSave = afterSaveTemp;
+			done();
 		});
-	});
-
-	describe('__private', () => {
-		describe('afterSave', () => {
-			it('should call library.bus.message');
-
-			it('should call library.bus.message with "transactionsSaved"');
-
-			it('should call library.bus.message with block.transactions');
 
-			it(
-				'should call library.logic.transaction.afterSave for every block.transaction'
-			);
-
-			describe('when library.logic.transaction.afterSave fails', () => {
-				it('should call callback with error');
-			});
-
-			describe('when library.logic.transaction.afterSave succeeds', () => {
-				it('should call callback with error = undefined');
-
-				it('should call callback with result = undefined');
+		describe('when tx param is passed', () => {
+			let txStub;
+			beforeEach(done => {
+				txStub = {
+					blocks: {
+						save: sinonSandbox.stub(),
+					},
+					transactions: {
+						save: sinonSandbox.stub(),
+					},
+					batch: sinonSandbox.stub(),
+				};
+				done();
 			});
-		});
 
-		describe('applyTransaction', () => {
-			it('should call modules.transactions.applyUnconfirmed');
-
-			it('should call modules.transactions.applyUnconfirmed with transaction');
-
-			it('should call modules.transactions.applyUnconfirmed with sender');
-
-			it('should call modules.transactions.applyUnconfirmed with callback');
+			describe('tx.batch', () => {
+				describe('when fails', () => {
```

Comment:
Please change here and in the following examples from:
```
describe('tx.batch')
   describe('when fails')
describe('tx.batch')
   describe('when succeeds')
```
to the form which is used in the rest of the tests to stay consistent:
```
describe('when tx.batch fails')
describe('when tx.batch succeeds)
```

Reply:
The pattern is
```
describe('tx.batch')
    describe('when fails)
    describe('when succeeds')
```
It was used in all the test from `modules/blocks` folder, should I change all of them?

Reply:
There is the only`process.js` file that conforms to this new pattern. `api.js`, `utils.js`, `verify.js` are not using them. Please change the current file only for now, we can change the `process.js` file in another PR.